collector/src/main.rs

```diff
@@ -745,137 +686,196 @@ fn main() {
     }
 }
 
-fn main_result() -> anyhow::Result<i32> {
-    env_logger::init();
+#[derive(Debug, clap::Parser)]
+#[clap(about, version, author)]
+struct Cli {
+    #[clap(subcommand)]
```

Comment:
The [migration guide](https://github.com/clap-rs/clap/blob/master/CHANGELOG.md) recommends having
```rust
#[test]
fn verify_app() {
    use clap::IntoApp;
    Cli::into_app().debug_assert()
}
```

Clap assumes most errors are developer errors and asserts, rather than reporting them through `Result`.  By default, clap lazily checks subcommands.  This test will ensure they are eagerly tested, so you get feedback earlier than actually running your binary for each case.

Reply:
Good suggestion, thanks!